### PR TITLE
Add complete price function to result

### PR DESCRIPTION
**Pull request by jondoe1337**
Fixes requested review changes from 3.5.19 to show complete price function

### CODE REVIEW

File: validator-auction-rest-server/lib/EthEventsClient.js [hunk 1]
```diff
@@ -30,12 +30,19 @@ export default class EthEventsClient {
             this.getAuctionDeploymentParameters()
         ])
 
+        if(!deploymentParams || !auctionStart) {
```
Comment (berndbohmeier) [inline validator-auction-rest-server/lib/EthEventsClient.js]: If the auction has not started yet, we can still already show some infos about the auction, so I think we should include all information, which are available even before the start of the auction (number of slots, length, start price )

File: validator-auction-rest-server/lib/EthEventsClient.js [hunk 2]
```diff
@@ -193,6 +200,22 @@ export default class EthEventsClient {
         return startPriceWEI.mul(_1.add(t)).div(_1.add(t).add(decay))
     }
 
+    static calculateAllSlotPrices(startInSeconds, durationInDays, startPriceWEI, ticks = 312) {
```
Comment (berndbohmeier) [inline validator-auction-rest-server/lib/EthEventsClient.js:170]: We do you do this on the server? I think this can be done on the client.

**Review by berndbohmeier — APPROVED** [review]
Can be merged like this, but I feel a lot of the graph calculation should be on the client. The server should just give back the current status of the auction and the parameter. I think it is weird that the server calculates all the points for the price function on the server for every visitor.

File: validator-auction-rest-server/lib/EthEventsClient.js (hunk 2 above)
Comment (jondoe1337) [reply validator-auction-rest-server/lib/EthEventsClient.js:170]: That would mean to have business logic in the client and also to use another library (BN.js) in the client. Also you would end with having duplicate code (price function) in the client and server.

Comment (berndbohmeier) [reply validator-auction-rest-server/lib/EthEventsClient.js:170]: I think BN should only be on the client and the price function should also only be on the client. I would not duplicate that business logic but only having it on the client. There is no need doing it on the server. We can discuss this after the daily.

Comment (berndbohmeier) [reply validator-auction-rest-server/lib/EthEventsClient.js:170]: Team decided this is necessary to change at the moment.

File: validator-auction-rest-server/lib/EthEventsClient.js [hunk 3]
```diff
@@ -19,31 +19,37 @@ export default class EthEventsClient {
...
-        result.currentPriceInWEI = EthEventsClient.getCurrentPriceAsBigNumber(auctionStart * 1000, currentBlockTime * 1000, deploymentParams.durationInDays, deploymentParams.startPrice).toString()
-        result.priceFunction = EthEventsClient.calculateAllSlotPrices(auctionStart, deploymentParams.durationInDays, deploymentParams.startPrice)
-        result.currentBlocktimeInMs = currentBlockTime * 1000
-        return result
+        const deploymentParams = this.getAuctionDeploymentParameters(allEvents)
+        const bids = this.getBids(allEvents)
+        const whitelistedAddresses = this.getWhitelistedAddresses(allEvents)
+        let auctionStart = this.getAuctionStartInSeconds(allEvents)
+        if (!auctionStart) {
+            auctionStart = currentBlockTime
```
Comment (berndbohmeier) [inline validator-auction-rest-server/lib/EthEventsClient.js]: What is this, why do we take the currentBlockTime as starting time if the auction did not start yet?

File: validator-auction-rest-server/lib/EthEventsClient.js [hunk 4]
```diff
@@ -19,31 +19,37 @@ export default class EthEventsClient {
...
+        }
+        const currentPriceInWEI = state === 'Finished' ? this.getClosingPrice(allEvents).toString() : EthEventsClient.getCurrentPriceAsBigNumber(auctionStart * 1000, currentBlockTime * 1000, deploymentParams.durationInDays, deploymentParams.startPrice).toString()
```
Comment (berndbohmeier) [inline validator-auction-rest-server/lib/EthEventsClient.js]: What do we show for the other states? For example for Failed?

File: validator-auction-rest-server/lib/EthEventsClient.js (hunk 3 above)
Comment (jondoe1337) [reply validator-auction-rest-server/lib/EthEventsClient.js]: The currentBlockTime is not rendered anywhere on the UI, it is just used for calculation of the price curve.

File: validator-auction-rest-server/lib/EthEventsClient.js (hunk 4 above)
Comment (jondoe1337) [reply validator-auction-rest-server/lib/EthEventsClient.js]: When status is "Failed", the price is hidden on the UI.
When status is "Finished", the last slot price (given from the event) is rendered in the UI and the description changes to Closing Price.

File: validator-auction-rest-server/lib/EthEventsClient.js (hunk 3 above)
Comment (berndbohmeier) [reply validator-auction-rest-server/lib/EthEventsClient.js]: The auction price is only meaningful if the auction is started, so we should only calculate it if the state it started. This is connected to my other comment then I guess.

File: validator-auction-rest-server/lib/EthEventsClient.js (hunk 4 above)
Comment (berndbohmeier) [reply validator-auction-rest-server/lib/EthEventsClient.js]: I would rather have the currentPrice then only calculated when it makes sense. That we dont show it means probably that the value does not make sense, so we should make that clear by also not calculating it here. 

So:
auction deployed: undefined
auction started: calculate price
auction finished: undefined
auction errored: undefined

for the closing price we can give it a new name.

File: validator-auction-static/js/chart.js [hunk 5]
```diff
@@ -80,64 +143,79 @@ function renderChart(slotPrice, bidPrice) {
...
+    Chart.defaults.global.defaultFontSize = 16
 }
 
 function getAuctionData() {
-    // TODO: Change URL
     $.ajax({
         url: 'http://localhost:8090/auction-summary',
         success: function (result) {
+            currentResult = result
+            if(result.auctionHasNotStarted) {
```
Comment (berndbohmeier) [inline validator-auction-static/js/chart.js]: Is this still up to date? Now we have `result.state` right?

**Review by berndbohmeier — APPROVED** [review]

**Pull request merged**